lth_reproduction/src/mnist.py

```diff
@@ -1,2 +1,124 @@
-print('test')
+"""
+MNIST Classification with LeNet
+"""
+from __future__ import print_function
+import argparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+import gin
+
+
+class LeNet(nn.Module):
```

Comment:
Put together this LeNet using some existing sources and the LTH paper - LTH specifically mentions in figure 2 that the convolutions are 3x3 and fc layers are 300,100,10 - so I put that in, but I am a little unsure if the rest of the architecture (convolution blocks in particular) line up with LTH's LeNet